src/rpc/RPCServer.ts

```diff
@@ -357,19 +360,19 @@ class RPCServer {
     this.activeStreams.add(handlerProm);
   }
 
-  @ready(new rpcErrors.ErrorRpcDestroyed())
+  @ready(new rpcErrors.ErrorRPCDestroyed())
   public addEventListener(
     type: 'error',
-    callback: (event: RPCErrorEvent) => void,
+    callback: (event: rpcUtils.RPCErrorEvent) => void,
     options?: boolean | AddEventListenerOptions | undefined,
   ) {
     this.events.addEventListener(type, callback, options);
   }
 
-  @ready(new rpcErrors.ErrorRpcDestroyed())
+  @ready(new rpcErrors.ErrorRPCDestroyed())
   public removeEventListener(
```

Comment:
Question... why didn't you just extend the `EventTarget`?

Reply:
I can, but i'd lose the type information for the handler. dealing with the raw event target with custom events with added data can be confusing and annoying.  This also allows for the `@ready` decorator which I'm not sure is strictly needed.

Reply:
I'm extending it in the QUIC code base. I'm sure it's possible to do this without it. Also possible to have an overloaded signature in the subclass I think. I'll try.

Reply:
I've changed to extension here and also simplified the event class itself. Alignment with js-quic is more important here. However subclasses cannot provide a more specialised signature unfortunately. Maybe there's a hack, but not important.